src/java.base/share/classes/jdk/internal/ref/PhantomCleanable.java

```diff
@@ -62,73 +63,28 @@ public abstract class PhantomCleanable<T> extends PhantomReference<T>
      * @param referent the referent to track
      * @param cleaner  the {@code Cleaner} to register with
      */
+    @SuppressWarnings("this-escape")
     public PhantomCleanable(T referent, Cleaner cleaner) {
         super(Objects.requireNonNull(referent), CleanerImpl.getCleanerImpl(cleaner).queue);
-        this.list = CleanerImpl.getCleanerImpl(cleaner).phantomCleanableList;
-        insert();
+        this.index = -1;
+        this.list = CleanerImpl.getCleanerImpl(cleaner).activeList;
+        list.insert(this);
```

Comment:
For consistency, maybe use "this.list.insert(this)", or remove "this" from the preceding two lines.

Reply:
Right, done.